src/wallet/rpcwallet.cpp

```diff
@@ -2086,6 +2066,107 @@ UniValue sendmany(const JSONRPCRequest& request)
     return wtx.GetHash().GetHex();
 }
 
+/*
+ * This function uses [legacy_sendmany] in the background.
+ * If any recipient is a shielded address, instead it uses [shieldedsendmany "from_transparent"].
+ */
+UniValue sendmany(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 2 || request.params.size() > 5)
+        throw std::runtime_error(
+            "sendmany \"\" {\"address\":amount,...} ( minconf \"comment\" includeDelegated )\n"
+            "\nSend to multiple destinations. Recipients are transparent or shielded PIVX addresses.\n"
+            "\nAmounts are double-precision floating point numbers.\n"
+            + HelpRequiringPassphrase() + "\n"
+
+            "\nArguments:\n"
+            "1. \"dummy\"               (string, required) Must be set to \"\" for backwards compatibility.\n"
+            "2. \"amounts\"             (string, required) A json object with addresses and amounts\n"
+            "    {\n"
+            "      \"address\":amount   (numeric) The pivx address (either transparent or shielded) is the key,\n"
+            "                                     the numeric amount in PIV is the value\n"
```

Comment:
no memo support?

Reply:
that's not contemplated in `sendmany`.

Reply:
it could be though, since we're recreating the request further down and passing it on. I'm ok with excluding it for now, was just checking if this was intentional.